scripts/base/protocols/smtp/files.zeek

```diff
@@ -19,10 +23,17 @@ export {
 	global describe_file: function(f: fa_file): string;
 }
 
+event smtp_mail_data_file(f: fa_file, c: connection)
+	{
+	c$smtp$mail_fuid = f$id;
+	}
+
 function get_file_handle(c: connection, is_orig: bool): string
 	{
+	# Adding mail_fuid here if set to ensure the top-level mail message
+	# and the first MIME attachment do not get the same fuid allocated.
 	return cat(Analyzer::ANALYZER_SMTP, c$start_time, c$smtp$trans_depth,
-	           c$smtp_state$mime_depth);
+	           c$smtp?$mail_fuid ? c$smtp$mail_fuid : "", c$smtp_state$mime_depth);
```

Comment:
Not sure if this is a great idea, but if `mime_depth` came as -1 for the RFC822 message, things would automartically work (also elsewhere where scripts are hooking into the file analysis for SMTP).

Reply:
Interesting. `mime_depth` is a `count` today and documented to track the number of MIME encoded files, so that feels a bit magic (granted, including the fuid does, too, but it's not changing existing semantics).